File: chainOfResponsability/pyhton/soporte/soporte_especializado.py
from soporte.soporte import Soporte

class SoporteEspecializado(Soporte):
    def manejar_solicitud(self, solicitud):
        if "complejo" in solicitud:
            return "Soporte Especializado: Resolviendo problema complejo"

        if self.proximo_nivel:
            return self.proximo_nivel.manejar_solicitud(solicitud)

        return "Ningún nivel de soporte puede manejar la solicitud"

File: chainOfResponsability/php/src/SoporteEspecializado.php
<?php
namespace SoporteTecnico;

class SoporteEspecializado implements Soporte {
    private ?Soporte $proximoNivel = null;

    public function setProximoNivel(Soporte $siguiente): void {
        $this->proximoNivel = $siguiente;
    }

    public function manejarSolicitud(string $solicitud): ?string {
        if (strpos($solicitud, 'complejo') !== false) {
            return "Soporte Especializado: Resolviendo problema complejo";
        }

        if ($this->proximoNivel !== null) {
            return $this->proximoNivel->manejarSolicitud($solicitud);
        }

        return "Ningún nivel de soporte puede manejar la solicitud";
    }
}